wolfcrypt/src/asn.c

```diff
@@ -24174,7 +24180,15 @@ static int EncodeCert(Cert* cert, DerCert* der, RsaKey* rsaKey, ecc_key* eccKey,
 
     /* CA */
     if (cert->isCA) {
-        der->caSz = SetCa(der->ca, sizeof(der->ca));
+        der->caSz = SetCa(der->ca, sizeof(der->ca), cert->isCA);
+        if (der->caSz <= 0)
+            return CA_TRUE_E;
+
+        der->extensionsSz += der->caSz;
+    }
+    /* if isCA is false but Basic Constraint is set */
+    else if (cert->extBasicConstSet) {
```

Comment:
The body of the if and else if look to be the same. Can we just have one conditional, cert->isCa || cert->extBasicConstSet, then? Note that we probably shouldn't return CA_TRUE_E in the case where isCa is false. CA_TRUE_E is an oddly specific error. I'm surprised it exists.

Reply:
Good point on the if else. Returning CA_TRUE_E isn't ideal but David didn't think it was worth it to add a new error code